datalab/mlalpha/_confusion_matrix.py

```diff
@@ -11,57 +11,99 @@
-      counts: a list of count for each (predicted, true) combination.
+      input_csv: Path to a Csv file (with no header). Can be local or GCS path.
+      headers: Csv headers. If present, it must include 'target' and 'predicted'.
+      schema_file: Path to a JSON file containing BigQuery schema. Used if "headers" is None.
+    Returns:
+      A ConfusionMatrix that can be plotted.
+    Raises:
+      ValueError if both headers and schema_file are None, or it does not include 'target'
+          or 'predicted' columns.
+    """
 
-    Raises: Exception if predicted_labels, true_labels, and counts are not of the same size
+    if headers is not None:
+      names = headers
+    elif schema_file is not None:
+      with ml.util._file.open_local_or_gcs(schema_file, mode='r') as f:
+        schema = json.load(f)
+      names = [x['name'] for x in schema]
+    else:
+      raise ValueError('Either headers or schema_file is needed')
+    with ml.util._file.open_local_or_gcs(input_csv, mode='r') as f:
+      df = pd.read_csv(f, names=names)
+    if 'target' not in df or 'predicted' not in df:
+      raise ValueError('Cannot find "target" or "predicted" column')
+
+    labels = sorted(set(df['target']) | set(df['predicted']))
+    cm = confusion_matrix(df['target'], df['predicted'], labels=labels)
+    return ConfusionMatrix(cm, labels)
+
+  @staticmethod
+  def from_bigquery(sql):
+    """Create a ConfusionMatrix from a BigQuery table or query.
+    Args:
+      sql: Can be one of:
+          A SQL query string.
+          A SQL Query module defined with '%%sql --name [module_name]'.
+          A Bigquery table.
+      The query results or table must include "target", "predicted" columns.
+    Returns:
+      A ConfusionMatrix that can be plotted.
+    Raises:
+      ValueError if query results or table does not include 'target' or 'predicted' columns.
     """
-    if len(predicted_labels) != len(true_labels) or len(true_labels) != len(counts):
-      raise Exception('The input predicted_labels, true_labels, counts need to be same size.')
-    self._all_labels = list(set(predicted_labels) | set(true_labels))
-    data = []
-    for value in self._all_labels:
-      predicts_for_current_true_label = \
-          {p: c for p, t, c in zip(predicted_labels, true_labels, counts) if t == value}
-      # sort by all_values and fill in zeros if needed
-      predicts_for_current_true_label = [predicts_for_current_true_label.get(v, 0)
-          for v in self._all_labels]
-      data.append(predicts_for_current_true_label)
-    self._data = data
+
+    query, _ = data.SqlModule.get_sql_statement_with_environment(sql, {})
+    sql = ('SELECT target, predicted, count(*) as count FROM (%s) group by target, predicted'
+        % query.sql)
+    df = bq.Query(sql).results().to_dataframe()
+    labels = sorted(set(df['target']) | set(df['predicted']))
+    labels_count = len(labels)
+    df['target'] = [labels.index(x) for x in df['target']]
```

Comment:
We cannot. Scikit-learn's confusion-matrix constructor only accepts a list of "target" and a list of "predicted" (so it doesn't work with large data). It does not accept aggregated data.